docs/hazmat/primitives/asymmetric/ec.rst

```diff
@@ -122,6 +122,43 @@ Elliptic Curve Signature Algorithms
         :returns: A new instance of a :class:`EllipticCurvePublicKey`
             provider.
 
+Elliptic Curve Key Exchange algorithm
+-------------------------------------
+
+.. class:: ECDH(algorithm)
```

Comment:
One problem with that is that you need to pass the curve if you don't pass in a private key. Here's the API I previously suggested which is along these lines:

```python
class ECDH:
    def __init__(self, curve):
        """Construct an ECDH private key"""

    def get_public(self):
        """Return the ECDH public key to send in the handshake"""

    def get_shared(self, public):
        """Return the shared key given the peer's public key"""
```

Instead of having a member which lets you access the private key, there's a method to return the public key corresponding to it, since that's what you need to give to the peer.

To allow an existing private key to be used, that could be passed as another argument to the constructor, though there would need to be a check that the curve & private key matched if we allow both to be passed in.